pkg/inventory/inventory-client.go

```diff
@@ -102,9 +110,13 @@ func (cic *ClusterInventoryClient) Merge(localInv InventoryInfo, objs object.Obj
 			return nil, err
 		}
 		klog.V(4).Infof("creating initial inventory object with %d objects", len(objs))
-		if err := cic.createInventoryObj(invInfo, dryRun); err != nil {
+		var createdObj *unstructured.Unstructured
+		if createdObj, err = cic.createInventoryObj(invInfo, dryRun); err != nil {
 			return nil, err
 		}
+		if createdObj != nil {
```

Comment:
is this ever nil after success? I think we can assume it's non-nil, right?

Reply:
same question for apply.

Reply:
In the unit test, some functions that are mocked doesn't match the Get/Update behaviors, but I haven't figured out which part to change.

Reply:
If the mock doesn't represent reality, we should fix the mock.

I think we can safely assume the object is non-nil if the error is nil, except in the DELETE implementation, where the object is not returned from the DynamicClient, because it is inconsistently returned from the RESTClient underneath.

Reply:
How about adding a TODO for this and fix it afterwards? I can sync up with Morten to learn more clue for this, but I don't think we want to block the change for this.

Reply:
fair enough.

Reply:
Added the TODO.

Reply:
Fixed by https://github.com/kubernetes-sigs/cli-utils/pull/545